controllers/components/datasciencepipelines/datasciencepipelines_controller.go

```diff
@@ -19,40 +19,65 @@ package datasciencepipelines
 import (
 	"context"
 
-	"k8s.io/apimachinery/pkg/runtime"
+	securityv1 "github.com/openshift/api/security/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 
 	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/updatestatus"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
 )
 
-// DataSciencePipelinesReconciler reconciles a DataSciencePipelines object.
-type DataSciencePipelinesReconciler struct {
-	client.Client
-	Scheme *runtime.Scheme
-}
+var (
+	DefaultPath = odhdeploy.DefaultManifestPath + "/" + componentsv1.DataSciencePipelinesComponentName + "/openshift"
+)
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the DataSciencePipelines object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *DataSciencePipelinesReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// TODO(user): your logic here
-
-	return ctrl.Result{}, nil
-}
+func NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ComponentReconcilerFor[*componentsv1.DataSciencePipelines](
+		mgr,
+		componentsv1.DataSciencePipelinesInstanceName,
+		&componentsv1.DataSciencePipelines{},
+	).
+		// customized Owns() for Component with new predicates
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Secret{}).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&appsv1.Deployment{}, builder.WithPredicates(resources.NewDeploymentPredicate())).
+		Owns(&securityv1.SecurityContextConstraints{}).
+		Watches(&extv1.CustomResourceDefinition{}). // call ForLabel() + new predicates
```

Comment:
Doing a very raw check (I should improve it), there are additional resources that may be require to own/watch (i.e. `Service` and `ServiceMonitor` at the bottom):

```
➜ kustomize build opt/manifests/data-science-pipelines-operator/overlays/odh | yq eval-all '[.apiVersion, .kind]'          
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- apiextensions.k8s.io/v1
- CustomResourceDefinition
- v1
- ServiceAccount
- v1
- ServiceAccount
- rbac.authorization.k8s.io/v1
- Role
- rbac.authorization.k8s.io/v1
- Role
- rbac.authorization.k8s.io/v1
- ClusterRole
- rbac.authorization.k8s.io/v1
- ClusterRole
- rbac.authorization.k8s.io/v1
- ClusterRole
- rbac.authorization.k8s.io/v1
- ClusterRole
- rbac.authorization.k8s.io/v1
- ClusterRole
- rbac.authorization.k8s.io/v1
- ClusterRole
- rbac.authorization.k8s.io/v1
- ClusterRole
- rbac.authorization.k8s.io/v1
- ClusterRole
- rbac.authorization.k8s.io/v1
- RoleBinding
- rbac.authorization.k8s.io/v1
- RoleBinding
- rbac.authorization.k8s.io/v1
- ClusterRoleBinding
- rbac.authorization.k8s.io/v1
- ClusterRoleBinding
- rbac.authorization.k8s.io/v1
- ClusterRoleBinding
- v1
- ConfigMap
- v1
- ConfigMap
- v1
- ConfigMap
- v1
- Service
- apps/v1
- Deployment
- monitoring.coreos.com/v1
- ServiceMonitor
```